src/chainstate/stacks/boot/pox-3.clar

```diff
@@ -0,0 +1,1316 @@
+      (try! (minimal-can-stack-stx pox-addr new-total-locked first-increase-cycle (+ u1 (- last-increase-cycle first-increase-cycle))))
+
+      ;; register the PoX address with the amount stacked via partial stacking
+      ;;   before it can be included in the reward set, this must be committed!
+      (add-pox-partial-stacked pox-addr first-increase-cycle cycle-count increase-by)
+
+      ;; stacking-state is unchanged, so no need to update
+
+      ;; return the lock-up information, so the node can actually carry out the lock.
+      (ok { stacker: stacker, total-locked: new-total-locked}))))
+
+;; As a delegator, extend an active stacking lock, issuing a "partial commitment" for the
+;;   extended-to cycles.
+;; *New in Stacks 2.1*
+;; This method extends `stacker`'s current lockup for an additional `extend-count`
+;;    and partially commits those new cycles to `pox-addr`
+(define-public (delegate-stack-extend
+                    (stacker principal)
+                    (pox-addr { version: (buff 1), hashbytes: (buff 32) })
+                    (extend-count uint))
+    (let ((stacker-info (stx-account stacker))
+          ;; to extend, there must already be an entry in the stacking-state
+          (stacker-state (unwrap! (get-stacker-info stacker) (err ERR_STACK_EXTEND_NOT_LOCKED)))
+          (amount-ustx (get locked stacker-info))
+          (unlock-height (get unlock-height stacker-info))
+          ;; first-extend-cycle will be the cycle in which tx-sender *would have* unlocked
+          (first-extend-cycle (burn-height-to-reward-cycle unlock-height))
+          (cur-cycle (current-pox-reward-cycle))
+          ;; new first cycle should be max(cur-cycle, stacker-state.first-reward-cycle)
+          (cur-first-reward-cycle (get first-reward-cycle stacker-state))
+          (first-reward-cycle (if (> cur-cycle cur-first-reward-cycle) cur-cycle cur-first-reward-cycle)))
+
+     ;; must be called with positive extend-count
+     (asserts! (>= extend-count u1)
+               (err ERR_STACKING_INVALID_LOCK_PERIOD))
+
+     (let ((last-extend-cycle  (- (+ first-extend-cycle extend-count) u1))
+           (lock-period (+ u1 (- last-extend-cycle first-reward-cycle)))
+           (new-unlock-ht (reward-cycle-to-burn-height (+ u1 last-extend-cycle))))
+
+      ;; first cycle must be after the current cycle
+      (asserts! (> first-extend-cycle cur-cycle) (err ERR_STACKING_INVALID_LOCK_PERIOD))
+      ;; lock period must be positive
+      (asserts! (> lock-period u0) (err ERR_STACKING_INVALID_LOCK_PERIOD))
+
+      ;; must be called directly by the tx-sender or by an allowed contract-caller
+      (asserts! (check-caller-allowed)
+        (err ERR_STACKING_PERMISSION_DENIED))
+
+      ;; stacker must not be directly stacking
+      (asserts! (is-eq (len (get reward-set-indexes stacker-state)) u0)
+                (err ERR_STACKING_NOT_DELEGATED))
+
+      ;; stacker must be delegated to tx-sender
```

Comment:
;; stacker STX must have been locked initially by tx-sender

This implies that changing the pool requires 1 cool down cycle! This should go somewhere into the change log of this contract.

Reply:
This is a problem for sBTC mini. The current design requires that users can switch pools each cycle.